### ISSUE TITLE
Unable to open previously downloaded audiobook after last testflight update

### ISSUE THREAD

**Issue by UnixNight** [advplyr/audiobookshelf-app#953]
### Steps to reproduce

1. Have downloaded audiobook on v0.9.66
2. update to 0.9.67
3. open downloaded book

### Expected behaviour
- Audiobook should open as expected

### Actual behaviour
- White screen with the error "Attempted to assign to readonly property"

### Environment data

Audiobookshelf Version: 2.5.0 docker

- [ ] Android App?
- [x] iOS App?

#### Android Issue
Android version:

Device model:

Stock or customized system:

#### iOS Issue
iOS Version: 17.1.1

iPhone model: iPhone 14 Pro Max

**Comment by nichwall** [comment]
Does downloading the book again fix it?

**Comment by UnixNight** [comment]
Downgrading will just repeatedly crash the app unfortunately

**Comment by advplyr** [comment]
Can you delete the item from the downloads page?

**Comment by advplyr** [comment]
I have a bunch of books downloaded on 0.9.66 and some podcast episodes and I couldn't reproduce this so this must be a specific case. I couldn't see any assignments being made to readonly properties in the update

**Comment by UnixNight** [comment]
> Can you delete the item from the downloads page?

I was able to view the files from the download page but accessing the book from the main view was still throwing an error

> Does downloading the book again fix it?

I misread this. Redownloading after deleting it from the Download page seems to have resolved this issue. Weird

**Comment by advplyr** [comment]
Someone else mentioned this in Discord so unless that was you then you aren't the only one this happened to. I haven't been able to find a cause and so far it's too rare to have anything to go off of. Will re-open if more cases come up

**Comment by UnixNight** [comment]
That wasn't me. Hopefully it was just a weird edge case

**Comment by advplyr** [comment]
I happened to come across this issue on a device that had a downloaded book that was downloaded on server v2.2.23 or below. Since it was downloaded on v2.2.23 or below it was using the old id before we switched to using UUIDs. This caused the issue.

The reason this is rare is because the item will have to have been downloaded on server <= v2.2.23 and you will have to be running a server now that is >= v2.3.0. This will be fixed in the next release but I still think users should re-download to get the UUIDs stored locally.
